### PR TITLE
First try with type modeling.

### PR DESCRIPTION
**Pull request by Yamakaky**
The last five types are not done.

### CODE REVIEW

File: rspirv/lib.rs [hunk 1]
```diff
@@ -34,5 +34,6 @@ extern crate spirv_headers as spirv;
 pub mod binary;
 pub mod grammar;
 pub mod mr;
+pub mod ir;
```
Comment (Yamakaky) [inline rspirv/lib.rs]: it was for the "unused" warning ^^

Comment (antiagainst) [inline rspirv/lib.rs]: I see. Feel free to `#![allow(dead_code)]`.

**Comment by antiagainst** [comment]
Hi @Yamakaky, are you still interested in doing this?

**Comment by Yamakaky** [comment]
I started a lot of projects at the same time, so this one use paused. I managed to genre cendre accepted as a semester project in Computer Graphics, so I'll go back to it ;)

**Comment by antiagainst** [comment]
@Yamakaky: Awesome! Enjoy the hacking! ;-P

**Comment by antiagainst** [comment]
@kvark: Yep, agreed to what you said. I actually started prototyping using enums, see the `rspirv/sr/` directory, but pretty much preliminary at the moment.

I'll close this pull request since it has been stale for quite some time and the functionality proposed here is basically implemented in `rspirv/sr/`.

**Pull request closed without merging**